AWSTranscribeStreaming/AWSTranscribeStreamingService.m

```diff
@@ -309,15 +318,16 @@ - (AWSTask *)setUpWebsocketForRequest:(NSDictionary *)requestParams {
 
         NSURL *websocketURL = task.result;
         NSURLRequest *urlRequest = [NSURLRequest requestWithURL:websocketURL];
+        if (!self.webSocketProvider) { //if it wasn't initialized aka nothing was passed in then initialize it with socket rocket adaptor
+            self.webSocketProvider = [[AWSSRWebSocketAdaptor alloc] initWithURLRequest:urlRequest];
+            [self updateWebSocketDelegate];
```

Comment:
is `setUpWebsocketForRequest` client facing, called by the developer? if any type of websocket provider is instantiated with an URL then this method can have an overload to take in a webSocket provider. 

what happens in Amplify? 
1. init with websocket provider
2. call setUpWebsocketForRequest? urlRequest is never passed to native websocket provider. connect should fail

Reply:
taking a look at this more closely, looks like 
`- (void)startTranscriptionWSS:(AWSTranscribeStreamingStartStreamTranscriptionRequest *)request` is the client facing method.

1. startTranscriptionWSS calls invokeRequestForWSS
2. invokeRequestForWSS calls setUpWebsocketForRequest

so, i think init() should still take in websocketProvider, so perhaps it is just missing setting the URL before connecting for the passed in websocket.

Reply:
hm not sure what you're suggesting here? can you elaborate. the second comment here is correct in the order. this initWithURL is how socket rocket class is initalized and we can't init before we get a presigned url.

Reply:
the question would be, how does the native websocket get `urlRequest` passed to it before calling `connect`?

Reply:
if you look at this code
```
NSURLRequest *urlRequest = [NSURLRequest requestWithURL:websocketURL];
        if (!self.webSocketProvider) { //if it wasn't initialized aka nothing was passed in then initialize it with socket rocket adaptor
            self.webSocketProvider = [[AWSSRWebSocketAdaptor alloc] initWithURLRequest:urlRequest];
            [self updateWebSocketDelegate];

        }

        //Open the web socket
        [self.webSocketProvider connect];
```
it says that if the webSocketProvider was initialized, we connect. but i believe connect will fail because the webSocketURL is never passed to the initialized webSocketProvider

Reply:
ya this is a good point. should probably add a set url on the protocol for a web socket so we can do exactly as you suggested.

Reply:
this is the problem wiht that suggestion
```
- (instancetype)initWithURLRequest:(NSURLRequest *)urlRequest {
    AWSDDLogVerbose(@"Initializing adaptor with url request %@", urlRequest);
    if (self = [super init]) {
        _webSocket = [[AWSSRWebSocket alloc] initWithURLRequest:urlRequest];

    }
    return self;
}
```

Reply:
the socket rocket class is instantiated with a url so we cant "init" the class until we have a url which we dont have until we've gotten the presigned url

Reply:
this has been removed entirely